ledger/accountdb.go

```diff
@@ -620,16 +631,19 @@ func (qs *accountsDbQueries) lookupCreator(cidx basics.CreatableIndex, ctype bas
 	return
 }
 
-// lookup looks up for a the account data given it's address. It returns the current database round and the matching
+// lookup looks up for a the account data given it's address. It returns the persistedAccountData, which includes the current database round and the matching
 // account data, if such was found. If no matching account data could be found for the given address, an empty account data would
 // be retrieved.
-func (qs *accountsDbQueries) lookup(addr basics.Address) (data basics.AccountData, dbRound basics.Round, err error) {
+func (qs *accountsDbQueries) lookup(addr basics.Address) (data persistedAccountData, err error) {
 	err = db.Retry(func() error {
 		var buf []byte
-		err := qs.lookupStmt.QueryRow(addr[:]).Scan(&dbRound, &buf)
+		var rowid sql.NullInt64
+		err := qs.lookupStmt.QueryRow(addr[:]).Scan(&rowid, &data.round, &buf)
 		if err == nil {
-			if len(buf) > 0 {
-				return protocol.Decode(buf, &data)
+			data.addr = addr
+			if len(buf) > 0 && rowid.Valid {
```

Comment:
it shouldn't. However, since the code here is using the `rowid.Int64`, it seems reasonable to test `rowid.Valid`. Do you have any reason to believe it wouldn't work ?

Reply:
I just thought it is a redundant check, and wanted to confirm. It won't hurt to be there, just that redundancy is a bit confusing.